botutils/botconfig/config.go

```diff
@@ -11,8 +11,34 @@ import (
 	"github.com/palantir/go-githubapp/githubapp"
 )
 
+//go:generate mockgen -destination os_mock_test.go -self_package github.com/solo-io/go-utils/botutils/botconfig -package botconfig_test github.com/solo-io/go-utils/osutils OsClient
+
 const (
 	DefaultBotCfg  = "/etc/solo-github-app/config.yml"
+
+	BotConfigEnvVar = "BOT_CONFIG"
+	WebhookSecretEnvVar = "WEBHOOK_SECRET"
+	PrivateKeyEnvVar = "PRIVATE_KEY_FILE"
+	IntegrationIdEnvVar = "INTEGRATION_ID"
+	InstallationIdEnvVar = "INSTALLATION_ID"
+)
+
+var (
+	FailedToReadFileError = func(err error, path string) error {
+		return errors.Wrapf(err, "failed reading file: %s", path)
+	}
+
+	FailedToParseConfigError = func(err error, path string) error {
+		return errors.Wrapf(err, "failed parsing configuration file: %s", path)
+	}
+
+	FailedToParseEnvVarError = func(err error, name, value string) error {
+		return errors.Wrapf(err, "error parsing %s environment variable value: %s", name, value)
+	}
+
+	MissingBotConfigValueError = func(name string) error {
+		return errors.Errorf("missing important bot config, use %s to provide it or include in config map", name)
+	}
 )
 
 type Config struct {
```

Comment:
For such complex config with multiple sources of truth, might be worth looking at this https://github.com/spf13/viper
As I understand it, it's made to handle situations like this.